prost-build/src/code_generator.rs

```diff
@@ -219,6 +223,30 @@ impl <'a> CodeGenerator<'a> {
         }
     }
 
+    fn append_container_attributes(&mut self, msg_name: &str) {
+        assert_eq!(b'.', msg_name.as_bytes()[0]);
+        for &(ref matcher, ref attribute) in &self.config.container_attributes {
+            // We abuse the field matcher to match only whole message paths, since the ~~~ field is
```

Comment:
this is kind of hacky.  I think it would be simpler to change the `match_field` method to the following signature:

```
pub fn match_ident(matcher: &str, msg: &str, field: Option<&str>) -> bool;
```

And then reuse it for types (which would pass `None` for the field), as well as fields.  Internally all it has to do is not push on to the paths vec when the arg is `None`.